### PR TITLE
Feature gate serving AWS S3 blobs from oci-proxy

### LINKED ISSUE #80: gate or temporarily compile out s3 redirect
since the s3 buckets are still not ready, we should not block #77 rollout to registry.k8s.io on production s3 bucket availability.

I didn't think it would take this long, so I went with the simpler option of not gating s3 redirection / maintaining two modes, figuring we'd promote to production soon enough, and the initial production app is *literally* just bog simple 3XX to k8s.gcr.io so we can get users on registry.k8s.io and shouldn't need patches.

since it is taking a while, we should make it possible to continue iterating on other aspects, like redirecting to the individual GCR regions backing k8s.gcr.io ourselves, and later artifact registry instances instaed.

### PR DESCRIPTION
**Pull request by upodroid**
Fixes #80

/cc @dims @ameukam 

~~Arnaud, we need to modify the sandbox instance of oci-proxy and set `SERVE_IMAGES_FROM_AWS` to `true`~~

### CODE REVIEW

**Comment by ameukam** [comment]
This may conflict with the purpose of https://github.com/kubernetes-sigs/oci-proxy/pull/82. 

Let's get feedback from folx working the AWS part before we move forward with this.

**Comment by upodroid** [comment]
I can switch the default to true instead of false and amend the environment variables for the prod deployment if the buckets aren't ready by then.

**Comment by ameukam** [comment]
> I can switch the default to true instead of false and amend the environment variables for the prod deployment if the buckets aren't ready by then.

In that case we should introduce this after we start serve from the S3 buckets. There is not specific value to introduce logic that already default the existing behavior.

/hold

**Comment by dims** [comment]
@upodroid comments for 2 minor things left inline, PTAL

**Comment by upodroid** [comment]
This can be merged now. Rebased with the changes from #87 and serving from AWS is automatically turned on.

**Comment by BobyMCbobs** [comment]
@upodroid, thanks for your changes! It looks fine.

My only question is: since the blobs are now synced, are we still needing to feature gate the S3 buckets?

Ref: #82

**Comment by BobyMCbobs** [comment]
> Yes, it is useful for anyone looking to use archeio as an image proxy.
> Also, if we ever have S3 issues, we have an easy way of turning it off in production.

@upodroid, sounds good to me!

/lgtm

**Review by jaypipes — APPROVED** [review]
Yeah, seems reasonable to me. ++

**Review by sftim — COMMENTED** [review]
Informally: LGTM

File: cmd/archeio/app/handlers.go [hunk 1]
```diff
@@ -97,47 +98,49 @@ func makeV2Handler(rc RegistryConfig, blobs blobChecker) func(w http.ResponseWri
...
-
-		// for blob requests, check the client IP and determine the best backend
-		clientIP, err := getClientIP(r)
-		if err != nil {
-			// this should not happen
-			klog.ErrorS(err, "failed to get client IP")
-			http.Error(w, err.Error(), http.StatusBadRequest)
-			return
-		}
+		if rc.ServeImagesfromAWS {
```
Comment (BenTheElder) [inline cmd/archeio/app/handlers.go:101]: TODO(minor): this is only set at startup and we're returning a request handler function, we can avoid the runtime check on each request.

**Pull request merged**